usage/usage.go

```diff
@@ -0,0 +1,135 @@
+package usage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go/v2"
+	"github.com/google/uuid"
+	"go.opencensus.io/metric/metricdata"
+	"go.opencensus.io/metric/metricexport"
+)
+
+// Options provides options for LogExporter
+type Options struct {
+	// ReportingInterval is a time interval between two successive metrics export.
+	ReportingInterval time.Duration
+}
+
+type Usage struct {
+	TimeStamp time.Time
+	Count     int64
+	Size      int64
+}
+
+type UsageCollector struct {
+	id             uuid.UUID
+	reader         *metricexport.Reader
+	ir             *metricexport.IntervalReader
+	initReaderOnce sync.Once
+	o              Options
+	db             clickhouse.Conn
+	dbName         string
+	tableName      string
+	usageParser    func(metrics []*metricdata.Metric) (map[string]Usage, error)
+	prevCount      int64
+	prevSize       int64
+}
+
+var CollectorID uuid.UUID
+
+func init() {
+	CollectorID = uuid.New()
```

Comment:
@nityanandagohain should we use existing otel-collector id rather than creating a new one. By default, an otel-collector id is created which is called service-instance-id which can be reused

Reply:
Wasn't able to get a way to extract the collector ID that is already allocated. Will try again.